src/core/symbology-ng/qgssymbollayer.cpp

```diff
@@ -421,7 +421,25 @@ void QgsMarkerSymbolLayer::startRender( QgsSymbolRenderContext& context )
 void QgsMarkerSymbolLayer::drawPreviewIcon( QgsSymbolRenderContext& context, QSize size )
 {
   startRender( context );
-  renderPoint( QPointF( size.width() / 2, size.height() / 2 ), context );
+  QgsPaintEffect* effect = paintEffect();
+  if ( effect && effect->enabled() )
+  {
+    static QPointF nullPoint( 0, 0 );
+
+    QPainter* p = context.renderContext().painter();
+    p->save();
+    p->translate( QPointF( size.width() / 2, size.height() / 2 ) );
+
+    effect->begin( context.renderContext() );
+    renderPoint( nullPoint, context );
```

Comment:
Forget it, it's to compensate for the translation in the painter I guess... Should that also be applied to line and polygon?